AppDB/appscale/datastore/zkappscale/transaction_manager.py

```diff
@@ -160,10 +161,14 @@ def _delete_counter(self, path):
       path: A string specifying a ZooKeeper path.
     """
     try:
-      self.zk_client.delete(path)
+      try:
+        self.zk_client.delete(path)
+      except NotEmptyError:
+        # Cross-group transaction nodes have a child node.
+        self.zk_client.delete(path, recursive=True)
```

Comment:
why not always make recursive True?

Reply:
If we don't expect the node to have children, the delete call is more efficient. A recursive delete first makes a get_children call: https://github.com/python-zk/kazoo/blob/4d268adf9837836f05dde5ec81be0d7bbd759e78/kazoo/client.py#L1329